bpf/bpf_host.c

```diff
@@ -1172,14 +1172,15 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, const bool from_host)
 		}
 # endif /* defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV6) */
 
-#ifdef ENABLE_WIREGUARD
+# ifdef ENABLE_WIREGUARD
 		if (!from_host) {
 			next_proto = ip6->nexthdr;
-			l4_off = ETH_HLEN + ipv6_hdrlen(ctx, &next_proto);
-			if (ctx_is_wireguard(ctx, l4_off, next_proto, ipcache_srcid))
+			hdrlen = ipv6_hdrlen(ctx, &next_proto);
+			if (unlikely(hdrlen < 0) &&
```

Comment:
```suggestion
			if (likely(hdrlen > 0) &&
```

Reply:
Note that the patch description in https://github.com/cilium/cilium/pull/35345/commits/2a015864889a365e4184607053cac99c31a68e2b would need to be adjusted for the current behavior.

Reply:
Thanks for both, just changed them accordingly.